app/webpacker/components/CompetitionsOverview/CompetitionsFilters.js

```diff
@@ -81,21 +81,23 @@ export function EventSelector({
   onEventSelection,
   eventList = WCA_EVENT_IDS,
   disabled = false,
+  maxEvents = WCA_EVENT_IDS.length,
 }) {
   return (
     <>
       <label htmlFor="events">
         {`${I18n.t('competitions.competition_form.events')}`}
         <br />
-        <Button primary type="button" size="mini" id="select-all-events" onClick={() => onEventSelection({ type: 'select_all_events' })}>{I18n.t('competitions.index.all_events')}</Button>
+        <Button disabled={disabled || selectedEvents.length >= maxEvents} primary type="button" size="mini" id="select-all-events" onClick={() => onEventSelection({ type: 'select_all_events' })}>{I18n.t('competitions.index.all_events')}</Button>
```

Comment:
Why is the select-all button disabled depending on how many events the user has selected. Should it only be disabled if the `eventList` length is longer than the max events?

(And even then, it might still be quickest to select all events then deselect those you don't want, even if that temporarily goes over the max?)

Reply:
yeah, I will just disable it from the start and add a text underneath it.